libs/remix-ai-core/src/agents/completionAgent.ts

```diff
@@ -1,23 +1,132 @@
-import * as fs from 'fs';
+import lunr from 'lunr';
 
-class CodeCompletionAgent {
-  private codebase: string[];
+interface Document {
+  id: number;
+  filename: string;
+  content: string;
+  identifier: number;
+}
+
+interface indexT{
+  isIndexed: boolean;
+  lastIndexedTime?: number;
+  reason?: string;
+}
+
+enum SupportedFileExtensions {
+  solidity = '.sol',
+  vyper = '.vy',
+  circom = '.circom',
+}
+
+export class CodeCompletionAgent {
+  props: any;
+  indexer: any;
+  Documents: Document[] = [];
+  INDEX_THRESHOLD = 0.1;
+  N_MATCHES = 1;
+  indexed: indexT = {
+    isIndexed: false,
+    lastIndexedTime: 0,
+    reason: 'Init',
+  };
+
+  constructor(props) {
+    this.props = props;
+    this.listenForChanges();
+    this.indexer =lunr(function () {
+      this.ref('id')
+      this.field('filename')
+      this.field('content')
+      this.field('Identifier');
+    });
+
+    // listen for file changes
+    this.props.on('fileManager', 'fileRemoved', (path) => { this.indexed.isIndexed = false; });
+
+    setInterval(() => {
+      this.indexWorkspace()
+    }, 60000)
+  }
+
+  listenForChanges() {
+    this.props.on('fileManager', 'fileAdded', (path) => { this.indexed = { isIndexed: false, reason:"fileAdded" } });
+    this.props.on('fileManager', 'fileRemoved', (path) => { this.indexed = { isIndexed: false, reason:"fileRemoved" } });
+    this.props.on('filePanel', 'workspaceCreated', () => { this.indexed = { isIndexed: false, reason:"workspaceCreated" } });
+    this.props.on('filePanel', 'workspaceRenamed', () => { this.indexed = { isIndexed: false, reason:"workspaceRenamed" }});
+    this.props.on('filePanel', 'workspaceDeleted', () => { this.indexed = { isIndexed: false, reason:"workspaceDeleted" } });
+  }
 
-  constructor(codebasePath: string) {
-    // git or fs
-    this.codebase = this.loadCodebase(codebasePath);
+  async getDcocuments() {
+    const documents: Document[] = [];
+    const dirs = await this.props.call('fileManager', 'dirList', '/');
```

Comment:
did you try to call `await plugin.call('fileManager', 'copyFolderToJson', '/')` if should returns a JSON value with all the files.